src/main/java/com/bringframework/registry/BeanDefinitionRegistryImpl.java

```diff
@@ -0,0 +1,51 @@
+package com.bringframework.registry;
+
+import com.bringframework.exceptions.BeanDefinitionDuplicateNameException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Base implementation of {@link BeanDefinitionRegistry} used to store {@link BeanDefinition} and
+ * retrieve them. This implementation is thread-safe.
+ */
+@Slf4j
+public class BeanDefinitionRegistryImpl implements BeanDefinitionRegistry {
+  private final Map<String, BeanDefinition> registry = new ConcurrentHashMap<>();
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void registerBeanDefinition(String name, BeanDefinition beanDefinition) {
+    if (registry.containsKey(name)) {
+      log.error("BeanDefinition with name {} already exists. Registry BeanDefinition class is {} "
+              + "and provided for creation is {}", name, registry.get(name).getBeanClass(),
+          beanDefinition.getBeanClass());
```

Comment:
I feel like logging an error and throwing exception is too much.
Would it be enough just throw an exception with comprehensive message ? What do you think ?

Reply:
Fixed